### ISSUE TITLE
Randomized items in marts+Evolution Stones[BUG]

### ISSUE THREAD

**Issue by Nagitox** [rotobash/pokemon-ngc-rando#25]
**Describe the bug**
Items are not randomized in marts, and Agate Mart doesn't sell Evolution Stones

**Settings You Used**
[Sett XD.pdf](https://github.com/rotobash/pokemon-ngc-rando/files/7979262/Sett.XD.pdf)

**Expected behavior**
No crashes

**Screenshots**

![Capture d’écran (83)](https://user-images.githubusercontent.com/97539611/151987544-1e750d3b-1594-4fcb-8c66-9371e0f8f701.png)
![Capture d’écran (84)](https://user-images.githubusercontent.com/97539611/151987550-54e05ce8-a23d-49e3-a298-2fefc0d39e14.png)
![Capture d’écran (85)](https://user-images.githubusercontent.com/97539611/151987551-2658b64e-7d7f-4327-8a4d-38e2a3226ef7.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Additional context**
Everything else works fine, even items and quantities in overworld.

**Comment by rotobash** [comment]
Hey there, thanks for reporting this. Sorry for the delay in responding.

This looks like an issue with the PAL version (and probably JP but unconfirmed), most likely the offsets for items aren't exactly the same as in the US version. I will look into this and update you when I've found the proper offsets.

Cheers!

**Comment by Nagitox** [comment]
Thanks for the answer ;)

I just checked the first mart, with the US version, it works fine (I could not check Agate Mart yet), 
So your theory about the bug being an issue on the PAL version might be right

**Comment by rotobash** [comment]
Hi there, sorry for the wait. I wanted to give you an update on this.

So, in XD (probably Colosseum as well) the item tables and the item list that mart's use live in two different files for some reason. However, this file for marts (`pocket_menu`) contains a lot of the text used in the game (e.g. item descriptions, names, etc.). In the PAL version, there is a separate version for each language which wasn't seen by the randomizer. 

TLDR: the English file was being randomized as intended, the other languages were left untouched.

I thought this would be pretty easy to fix now that I know what's happening. Which it was, however, there's a bit of an issue. For some reason randomizing the items in the mart changes the amount of items that mart sells every time. I'm not sure why this happens tbh, I'll have to look into it a bit more. Didn't realize you can't just switch languages on an existing save which kinda sucked. Fortunately, now I have a save file in another language I can use for testing so it'll be easier to test (man, was it tricky getting that far when you don't speak the language. I kept getting dumpstered in battle because I couldn't tell what moves I had or were being used lol). 

Thanks again for the report, I would have never caught this otherwise 😅

**Comment by Nagitox** [comment]
Thanks for updating me on this point, I now know where you are in fixing the issue (It is strage though that the randomizer also change the amounts, I guesse it might be a difference inthe way of making the Code.

I know what you mean when you say you have trouble with moves with another languages, it happened as well when I tried to see if randomize marts worked in English, but since I'm used to play games in English, I guess I didn't struggled to much.

No need to thanks for reporting this, it's been a while I wanted a XD randomizer, so I thought I had to report an issue, to make it perfect 
Thanks again for Updating me on this point, I guesse I'll try to play the game in English, waiting for the issue to be solved ;)

**Comment by Nagitox** [comment]
I noticed after MT Battle, when Agate Marts is supposed to add Pokeball, its stock completely change, and, unfortunatelly, doesn't guarantee Evolution stones after the shift (I luckily can only buy Water stones and Everstones)

![Capture d’écran (114)](https://user-images.githubusercontent.com/97539611/157308891-e0130264-1466-4e14-9fd9-ec5e5d4827da.png)
![Capture d’écran (115)](https://user-images.githubusercontent.com/97539611/157308904-fab0b4a6-1caf-4b5d-a751-85d1ae7a865a.png)
 
 I know it's common for pokemon games to completely change stock when a single item is added, or change, it does the same thing with other randomizers, so I guesse there should be somewhere in the code the Inventory for the Mart after it add Pokeballs

**Comment by rotobash** [comment]
Hey, I fixed items on PAL and the missing evolution stones in Agate mart

Unfortunately, the PAL marts don't seem to add evolution stones even though the items are randomized. I have a suspicion it added them somewhere but I'm not sure where. The martsin code are only identified by a number so it's hard to match them with the in-game mart it's for. I'm keeping this issue open as a reminder to look into this more, buit the other issues you mentioned should be fixed now.

As always, feel free to let me know if you run into anything else. Cheers!
